odl/discr/grid.py

```diff
@@ -237,7 +234,7 @@ def shape(self):
     @property
     def size(self):
         """Total number of grid points."""
-        return np.prod(self.shape)
+        return 0 if self.shape == () else np.prod(self.shape)
```

Comment:
Not sure if it depends on the Numpy version, but I'll write a comment in the code.